p2p/upgrade.go

```diff
@@ -384,9 +384,11 @@ func (fh *Host) Start() error {
 	}
 	if err := fh.Network().Listen(fh.cfg.Listen...); err != nil {
 		fh.Network().Close()
-		return err
+		return fmt.Errorf("p2p: can't listen: %w", err)
+	}
+	if err := fh.discovery.Start(); err != nil {
+		return fmt.Errorf("error starting discovery: %w", err)
```

Comment:
NIT: I usually just add some context around when the error happened and leave "error/failed" for the log message. Otherwise, there bad-looking logs like "failed to X: error doing Y: error something: reason".
```suggestion
		return fmt.Errorf("starting discovery: %w", err)
```